### DISCUSSION TITLE
How do I work with Document fields in custom resolvers?

### DISCUSSION THREAD

**Discussion by WillSquire** [keystonejs/keystone#7144] [Questions] [answered]
I can't find documentation on the document field type. Want to cast `item` properties on `virtual`'s `resolve` to types, such as a document field to a document field type for safety. E.g.:

```ts
resolve(item, args, context, info) {
  const x = item.content; // <-- want to cast to a document type

  // ...
},
```

**Comment by WillSquire** [comment, marked as answer]
Found out it was already serialised to a `string`, but the type I wanted was in the `slate` package (i.e. the `Node` type). I wanted to calculate the reading time - if this helps anyone else, here's what I did:

```ts
readingTime: virtual({
  field: graphql.field({
    type: graphql.Int,
    resolve(item) {
      const content: Node[] = JSON.parse(item.content as string);
      const text = content.map((n) => Node.string(n)).join("\n");
      const { minutes } = readingTime(text);

      return Math.round(minutes);
    },
  }),
}),
```

It would be good to have some documentation on this.

**Reply by raveling** [reply]
@WillSquire thanks for the suggestion to improve the docs around this. We've added it to our TODO list of docs to  improve 🙇

**Reply by dcousens** [reply]
@WillSquire when https://github.com/keystonejs/keystone/pull/7733 (or similar) is merged, this shouldn't require you to cast your `item.content as string`.
